test/index.js

```diff
@@ -125,3 +125,32 @@ test('fail invalid signature', function(t) {
     t.end();
   }, 2000);
 });
+
+
+test('with express.js and body-parser incorrectly mounted', function(t) {
+  var express = require('express');
+  var app = express();
+  app.use(require('body-parser').json());
+  app.use(verifier);
+  var server = app.listen(3000);
+  app.get('/', function(req, res) {
+    res.status(200).send('ok');
+  });
+  var request = require('supertest');
+  request(server)
+    .post('/')
+    .send({ x: 1 })
+    .set('signaturecertchainurl', 'dummy')
+    .set('signature', 'aGVsbG8NCg==')
+    .end(function (err, res) {
+      t.equal(res.statusCode, 400);
+      t.deepEqual(res.body, {
+        "reason": "The raw request body is not available.",
```

Comment:
But that is cosmetic, the bug is that the request hangs not that the message is wrong.